pypesto/model_selection/misc.py

```diff
@@ -0,0 +1,295 @@
+        if par_id not in petab_problem.x_ids:
+            logger.info('%sWarning: parameter %s is not defined '
+                        'in PETab model. It will be ignored.',
+                        Fore.YELLOW,
+                        par_id)
+            # logger.info(Fore.YELLOW + f'Warning: parameter {par_id} is not '
+            #             'defined in PETab model. It will be ignored.')
+            continue
+        if not np.isnan(par_val):
+            petab_problem.parameter_df[ESTIMATE].loc[par_id] = 0
+            petab_problem.parameter_df[NOMINAL_VALUE].loc[par_id] = par_val
+            # petab_problem.parameter_df.lowerBound.loc[par_id] = float("NaN")
+            # petab_problem.parameter_df.upperBound.loc[par_id] = float("NaN")
+        else:
+            petab_problem.parameter_df[ESTIMATE].loc[par_id] = 1
+            # petab_problem.parameter_df.nominalValue.loc[par_id] = float(
+            # "NaN")
+
+    # Any parameter values in `x_guess` for parameters that are not estimated
+    # should be filtered out and replaced with
+    # - their corresponding values in `row` if possible, else
+    # - their corresponding nominal values in the `petab_problem.parameter_df`.
+    # TODO reconsider whether filtering is a good idea (x_guess is no longer
+    # the latest MLE then). Similar todo exists in
+    # `ModelSelectorMethod.new_model_problem`.
+    if x_guess is not None:
+        filtered_x_guess = []
+        for par_id, par_val in x_guess:
+            if petab_problem.parameter_df[ESTIMATE].loc[par_id] == 1:
+                filtered_x_guess.append(par_val)
+            else:
+                if par_id in row_parameters:
+                    filtered_x_guess.append(row_parameters[par_id])
+                else:
+                    filtered_x_guess.append(
+                        petab_problem.parameter_df[NOMINAL_VALUE].loc[par_id])
+        x_guesses = [filtered_x_guess]
+    else:
+        x_guesses = None
+
+    # chose standard objective in case none is provided
+    importer = PetabImporter(petab_problem)
+    if obj is None:
+        obj = importer.create_objective()
+    pypesto_problem = importer.create_problem(obj, x_guesses=x_guesses)
+
+    return pypesto_problem
+
+
+def aic(n_estimated: int, nllh: float) -> float:
+    """
+    Calculate the Akaike information criterion for a model.
+
+    Arguments
+    ---------
+    n_estimated:
+        The number of estimated parameters in the model.
+
+    nllh:
+        The negative log likelihood (e.g. the `optimize_result.list[0].fval`
+        attribute of the return value of `pypesto.minimize`).
+    """
+    return 2*(n_estimated + nllh)
+
+
+def bic(n_estimated: int, n_measurements: int, nllh: float):
+    """
+    Calculate the Bayesian information criterion for a model.
+    TODO untested
+
+    Arguments
+    ---------
+    n_estimated:
+        The number of estimated parameters in the model.
+
+    n_observations:
```

Comment:
should also contain the number of parameter priors, these are also considered data points.

Reply:
something to keep in mind :)